### PR TITLE
Filter tiles in reactor

### LINKED ISSUE #83: Filtering not applied when tile is part of build
When running a build in which the a tile is used and in the list of projects to be built, and the tile has some properties to be replaced, these are not replaced.

I has a look at the code and this is due to the fact that in TilesMavenLifecycleParticipant.resolveTile() there a piece of code that goes through the list of projects in the maven session and if one matches a tile, the tile.xml file is used directly rather than resolving the tile from the repo. This is fine for tiles that do not replace property values using filtering but breaks tiles that do. 

I have a project that has this situation. what I used to do before (with version 2.10) is first build the tile then build everything (including the tiles). However, with version 2.12, this breaks and I would have to separate my tiles from the rest of the code. I know I can solve this by breaking up the project into smaller projects, which will be done in due course, but then this pattern would not be needed anyway.

I tried (and managed) to add filtering during the lifecycle, however the code turned out quite ugly and requires a number of assumptions. I would suggest to
- either remove this part of the lifecycle, and let tiles only be resolved from the repo or
- do the filtering during the lifecycle as well (I have code for this which I can submit for opinions. I would rather not submit it as a pull request yet until we get an agreement on the assumptions)

### PR DESCRIPTION
**Pull request by bvella**
Extracted from #84. Fixes #83.

Perform maven standard filtering (replacing properties) in tiles in the same reactor build before injecting

### CODE REVIEW

File: src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
  * the License. You may obtain a copy of the License at
  *
- *	 http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (talios) [inline src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy]: Should be squashed into original commit.

File: src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy [hunk 2]
```diff
@@ -188,7 +189,7 @@ public class TilesMavenLifecycleParticipant extends AbstractMavenLifecyclePartic
 
 			// When resolving from workspace (e.g. m2e) we might receive the path to pom.xml instead of the attached tile
 			if (tileArtifact.file && tileArtifact.file.name == "pom.xml") {
-				tileArtifact.file = new File(tileArtifact.file.parent, FilteringHelper.TILE_POM)
+				tileArtifact.file = new File(tileArtifact.file.parent, TILE_POM)
```
Comment (talios) [inline src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy]: Squash into original commit.

File: src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy (hunk 1 above)
Comment (bvella) [reply src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy]: Pardon the stupid question, but how do I squash into the original commit? Do I squash into a new branch and submit as a new pr?

Comment (bvella) [reply src/test/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipantTest.groovy]: nevermind

File: src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy (hunk 2 above)
Comment (bvella) [reply src/main/groovy/io/repaint/maven/tiles/TilesMavenLifecycleParticipant.groovy]: done

**Comment by bvella** [comment]
Do you need anything else to merge this PR?

**Comment by talios** [comment]
Will check it out - probably not now we have updated the PR.

**Comment by bvella** [comment]
Hi @talios is there anything else you need to merge this PR?

**Comment by talios** [comment]
Have repulled the branch locally to test.... Sorry I didn't notice the rebase...

**Comment by talios** [comment]
When rebased on the current master, this branch currently failed on the `sitereporting-tiletest` integration test:

```
[ERROR] Internal error: java.lang.NullPointerException -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: java.lang.NullPointerException
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:120)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at org.apache.maven.model.plugin.DefaultReportingConverter.convert (DefaultReportingConverter.java:243)
    at org.apache.maven.model.plugin.DefaultReportingConverter.convert (DefaultReportingConverter.java:213)
    at org.apache.maven.model.plugin.DefaultReportingConverter.convertReporting (DefaultReportingConverter.java:140)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:479)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:432)
    at org.apache.maven.model.building.ModelBuilder$build$0.call (Unknown Source)
    at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall (CallSiteArray.java:47)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call (AbstractCallSite.java:116)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call (AbstractCallSite.java:136)
    at io.repaint.maven.tiles.TilesMavenLifecycleParticipant.thunkModelBuilder (TilesMavenLifecycleParticipant.groovy:471)
    at io.repaint.maven.tiles.TilesMavenLifecycleParticipant.orchestrateMerge (TilesMavenLifecycleParticipant.groovy:374)
    at io.repaint.maven.tiles.TilesMavenLifecycleParticipant.afterProjectsRead (TilesMavenLifecycleParticipant.groovy:318)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
```

I'll try set aside some time tonight to look at this, and if it passes we can merge it (finally).

**Pull request merged**